source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+allows people to get started without hosting a Chef-server.
+
+The Chef toolset also comes with the `knife` command, which is a great command
+line tool for interacting with the Chef-server. It also is a parallel-ssh tool,
+manipulates chef cookbooks, and can also launch ec2 (and other) instances. (did
+they take the kitchen-sink metaphor too far?)
+
+The [Chef Supermarket](https://supermarket.chef.io/) serves as the public
+registry for Chef cookbooks. There are not too many quality indicators to see,
+to help find which cookbooks are any good. The best metric I could see is just
+sorting by "followers". This is made up by the fact the fact that there is over
+a hundred [officially supported
+cookbooks](https://supermarket.chef.io/users/chef). 
+
+Probably the most difficult aspect of Chef for me to understand was how
+attributes interact.  This confusion probably most obvious when you look at
+Chef's [15 levels of attribute
+precedence](https://docs.chef.io/attributes.html#attribute-precedence). Seems
+to me that there should be a more obvious way for intent to flow, but I could
+be just spoiled by hiera.
+
+#### Review of the Chef-Sensu Cookbook
+
+The [Sensu Chef Cookbook](https://github.com/sensu/sensu-chef) is also a
+first-class citizen in the Sensu-world. The "native config language" of the
+main author of Sensu, [Sean Porter](https://github.com/portertech), is also
+"Chef". This gives a lot of credibility to the Cookbook, and shows in the
+[contributor page](https://github.com/sensu/sensu-chef/graphs/contributors).
+
+The Cookbook itself is feature complete, with recipes for installing and
+configuring all aspects of Sensu.
+
+The scope of the cookbook includes all Sensu related technologies, including
+RabbitMQ, Redis, and Uchiwa. It is certainly "batteries included" and on by
+default.  It even downloads and compiles [Redis from
+source](https://github.com/sensu/sensu-chef/issues/340) for you.
+
+Another example of this is the RabbitMQ module
+[setting](https://github.com/jjasghar/rabbitmq/pull/301)
+[Apt](https://github.com/jjasghar/rabbitmq/pull/302/files) attributes. Like the
+above Redis example, this behavior surprised me, but technically not related to
+the Sensu chef cookbook.
+
+At the same time, [wrapper
+cookbooks](https://github.com/portertech/chef-monitor) are recommended as a
+method to combine multiple cookbooks together in a coherent way. I think in
+general I just expected the wrapper cookbooks to do more and the main Sensu
+cookbook to do less.
+
+The cookbook does have an integration test suite, but it is not run via Travis.
+The code is under active development, and does multiple releases a year. It has
+native support for Chef [data bags](https://docs.chef.io/data_bags.html) for
+transporting the RabbitMQ SSL support, which is a nice touch. (Not tested in
+this review)
+
+### Ansible
+
+#### Ansible in General
+
+Ansible is a relative new-comer to the configuration-management space. Ansible
```

Comment:
Previously in the post you did not use a hyphen for 'configuration management'.

Sidenote: As a trend, I think you are more likely to hyphenate nouns when you have recently used a hyphen